packages/html/src/form/form-field.tsx

```diff
@@ -0,0 +1,92 @@
+import { classNames } from '../utils';
+
+export const FORMFIELD_CLASSNAME = 'k-form-field';
+
+export type KendoFormFieldProps = {
+    label?: null | string;
+    optional?: boolean;
+    component?: JSX.Element | string;
+    hint?: null | string;
+    error?: null | string;
+    info?: null | string;
+    disabled?: boolean;
+    alignText?: 'start' | 'end';
+    checkboxLabel?: null | string;
+    emptyLabel?: boolean;
+    dir?: "ltr" | "rtl";
+    fieldWrap?: boolean;
+};
+
+const defaultProps = {
+    alignText: 'start',
+    fieldWrap: true
+};
+
+export const FormField = (
+    props: KendoFormFieldProps &
+        React.HTMLAttributes<HTMLDivElement>
+) => {
+    const {
+        label,
+        optional,
+        component,
+        hint,
+        error,
+        info,
+        disabled,
+        alignText = defaultProps.alignText,
+        checkboxLabel,
+        emptyLabel,
+        dir,
+        fieldWrap = defaultProps.fieldWrap,
+    } = props;
+
+    const ConditionalWrapper = ({ condition, wrapper, children }) => (condition ? wrapper(children) : children);
+
+    return (
+        <div
+            className={classNames(
+                FORMFIELD_CLASSNAME,
+                props.className,
+                {
+                    'k-form-field-disabled': disabled
+                }
+            )} dir={dir}>
+            {label &&
+                            <label className="k-label k-form-label">
+                                {label}
+                                {optional && <span className="k-label-optional">(Optional)</span>}
+                                {info && <span className="k-field-info">(field info)</span>}
+                            </label>
+            }
+            {emptyLabel && <span className="k-label k-form-label k-label-empty"></span>}
+            <ConditionalWrapper
+                condition={fieldWrap}
+                wrapper={children => <div className="k-form-field-wrap">{children}</div>}>
+                <ConditionalWrapper
+                    condition={checkboxLabel}
+                    wrapper={children => <div className="k-form-field-checkbox-wrapper">{children}</div>}>
+                    {component}
+                    {checkboxLabel && <label className="k-checkbox-label">{checkboxLabel}</label>}
+                </ConditionalWrapper>
```

Comment:
I am not sure we need those conditionals as:

1) The `k-form-field-wrap` can be rendered everytime
2) The checkbox conditional is not needed, as the `<Checkbox />` spec already adds its required wrapper.